codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java

```diff
@@ -155,6 +155,18 @@ public ChannelFuture writeData(ChannelHandlerContext ctx, int streamId, ByteBuf
     }
 
     @Override
+    public ChannelFuture writeData(ChannelHandlerContext ctx, int streamId, ByteBuf data,
+            int padding, boolean endStream, ChannelPromise promise) {
+        return writeData(ctx, streamId, data, data.readableBytes(), padding, endStream, promise);
+    }
+
+    @Override
+    public ChannelFuture writeData(ChannelHandlerContext ctx, int streamId, FileRegion data,
+            int padding, boolean endStream, ChannelPromise promise) {
+        return writeData(ctx, streamId, data, (int) data.count(), padding, endStream, promise);
```

Comment:
As I suggested above we should pass a `length` to this method so we don't have to use `data.count()`.